### PR TITLE
feat: Add support for SSL enforcement modes

### LINKED ISSUE #58: Support for SSL enforcement modes for AlloyDB
AlloyDB now provides an ability to configure various SSL enforcement modes.
Public docs: [instance-ssl](https://cloud.google.com/alloydb/docs/instance-ssl)

Terraform AlloyDB instance resource is already supporting SSL. [Document](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/alloydb_instance#ssl_mode)

Currently we have to manually enable it from web interface.
Please can we have this issue resolved ?

### PR DESCRIPTION
**Pull request by rajc242**
Providing support for [SSL](https://cloud.google.com/alloydb/docs/instance-ssl) while deploying primary and read instances.

fix #58

### CODE REVIEW

**Comment by imrannayer** [comment]
@rajc242 thx for the PR. Can u plz fix lint issues.

File: variables.tf [hunk 1]
```diff
@@ -108,6 +108,7 @@ variable "primary_instance" {
     gce_zone          = optional(string)
     availability_type = optional(string)
     machine_cpu_count = optional(number, 2),
+    ssl_mode          = optional(string, "ALLOW_UNENCRYPTED_AND_ENCRYPTED"),
```
Comment (imrannayer) [inline variables.tf:111]: @rajc242 can you plz change this to

ssl_mode = optional(string)
require_connectors = optional(string)

File: variables.tf [hunk 2]
```diff
@@ -128,7 +129,8 @@ variable "read_pool_instance" {
     database_flags    = optional(map(string))
     availability_type = optional(string)
     gce_zone          = optional(string)
-    machine_cpu_count = optional(number, 2)
+    machine_cpu_count = optional(number, 2),
+    ssl_mode          = optional(string, "ALLOW_UNENCRYPTED_AND_ENCRYPTED"),
```
Comment (imrannayer) [inline variables.tf:133]: Change it to:
ssl_mode = optional(string)
require_connectors = optional(string)

File: main.tf [hunk 3]
```diff
@@ -155,6 +155,12 @@ resource "google_alloydb_instance" "primary" {
     cpu_count = var.primary_instance.machine_cpu_count
   }
 
+  client_connection_config {
```
Comment (imrannayer) [inline main.tf:158]: change following block to dynamic block

```
dynamic "client_connection_config" {
  for_each = coalesce(var.primary_instance.ssl_mode, var.primary_instance.require_connectors) == null ? [] : ["client_connection_config"]
    content {
      ssl_config {
        ssl_mode            = coalesce(var.primary_instance.ssl_mode)
        require_connectors  = coalesce(var.primary_instance.require_connectors)

      }
    }
}
```

File: main.tf [hunk 4]
```diff
@@ -181,6 +187,11 @@ resource "google_alloydb_instance" "read_pool" {
     cpu_count = each.value.machine_cpu_count
   }
 
+  client_connection_config {
```
Comment (imrannayer) [inline main.tf:190]: Make this also a dynamic block like [the one in ](https://github.com/GoogleCloudPlatform/terraform-google-alloy-db/pull/59/files#r1466969112)

File: variables.tf (hunk 1 above)
Comment (rajc242) [reply variables.tf]: Fixed. PLTL.

File: variables.tf (hunk 2 above)
Comment (rajc242) [reply variables.tf]: Fixed. PLTL.

File: main.tf (hunk 3 above)
Comment (rajc242) [reply main.tf]: Fixed but required_connectors order is bit different than suggested.

File: main.tf (hunk 4 above)
Comment (rajc242) [reply main.tf]: Fixed but required_connectors order is bit different than suggested.

**Pull request merged**